direct/io-bulkfs-parquet/src/main/java/cz/o2/proxima/direct/bulk/fs/parquet/ParquetFileFormat.java

```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright 2017-2021 O2 Czech Republic, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.o2.proxima.direct.bulk.fs.parquet;
+
+import com.google.common.base.Preconditions;
+import cz.o2.proxima.annotations.Internal;
+import cz.o2.proxima.direct.bulk.FileFormat;
+import cz.o2.proxima.direct.bulk.Path;
+import cz.o2.proxima.direct.bulk.Reader;
+import cz.o2.proxima.direct.bulk.Writer;
+import cz.o2.proxima.repository.AttributeFamilyDescriptor;
+import cz.o2.proxima.repository.EntityDescriptor;
+import cz.o2.proxima.storage.StreamElement;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.parquet.hadoop.ParquetOutputFormat;
+import org.apache.parquet.hadoop.metadata.CompressionCodecName;
+import org.apache.parquet.schema.MessageType;
+
+/** Bulk file format which stored data in Parquet format. */
+@Internal
+@Slf4j
+public class ParquetFileFormat implements FileFormat {
+
+  /** Config key specified compression used in parquet writer. */
+  public static final String PARQUET_CONFIG_COMPRESSION_KEY_NAME = ParquetOutputFormat.COMPRESSION;
+
+  /** Config key specified page and row group size in parquet writer. */
+  public static final String PARQUET_CONFIG_PAGE_SIZE_KEY_NAME = ParquetOutputFormat.PAGE_SIZE;
+
+  /**
+   * Config key used for attribute prefix - useful when attribute name collides with proxima fields
+   * defined as PARQUET_COLUMN_NAME_*
+   */
+  public static final String PARQUET_CONFIG_VALUES_PREFIX_KEY_NAME = "parquet.values.name.prefix";
+
+  /** Name for top level message in parquet schema. */
+  public static final String PARQUET_MESSAGE_NAME = "proxima-bulk";
```

Comment:
Maybe stream element?

```suggestion
  public static final String PARQUET_MESSAGE_NAME = "stream_element";
```